### PR TITLE
feat(strategy): add undefined strategy

### PR DESCRIPTION
**Pull request by qlaffont**
ref https://github.com/gcornut/valibot-json-schema/issues/14

### CODE REVIEW

File: src/toJSONSchema/schemas.ts [hunk 1]
```diff
@@ -156,4 +158,16 @@ export const SCHEMA_CONVERTERS: {
                 return { type: 'string', format: 'date-time' };
         }
     },
+    undefined(_, __, context) {
+        if (!context.undefinedStrategy) {
+            throw new Error('The "undefinedStrategy" option must be set to handle date validators');
```
Comment (gcornut) [inline src/toJSONSchema/schemas.ts]: A small oversight
```suggestion
            throw new Error('The "undefinedStrategy" option must be set to handle the `undefined` schema');
```

File: src/toJSONSchema/schemas.ts [hunk 2]
```diff
@@ -156,4 +158,16 @@ export const SCHEMA_CONVERTERS: {
...
+        }
+
+        switch (context.undefinedStrategy) {
+            case 'any':
+                return SCHEMA_CONVERTERS['any'];
+            case 'custom':
+                return {}
```
Comment (gcornut) [inline src/toJSONSchema/schemas.ts]: Looks like `custom` is exactly like `any`. Maybe we should only have the `any` strategy for now ?

**Review by gcornut — CHANGES_REQUESTED** [review]
I have two minor changes 
Also can you run `yarn lint --apply` before commit to properly format all the code ?

**Comment by qlaffont** [comment]
done ^^

**Review by gcornut — APPROVED** [review]

**Pull request merged**